packages/web3-core/src/web3_subscriptions.ts

```diff
@@ -74,7 +74,20 @@ export abstract class Web3Subscription<
 			err: Error | undefined,
 			data?: JsonRpcSubscriptionResult | JsonRpcNotification<Log>,
```

Comment:
Your changes indicate that `err` is/can be the expected return data. Does anyone @web3/web3-js know why `err` is the first argument for `messageListener`?

Reply:
From the [Examples section](https://eips.ethereum.org/EIPS/eip-1193#appendix-ii-examples) of EIP-1193, it would appear the a failure to subscribe would reject the promise, not that an error would be sent as a `message`

```typescript
// Example 4: Log new blocks
ethereum
  .request({
    method: 'eth_subscribe',
    params: ['newHeads'],
  })
  .then((subscriptionId) => {
    ethereum.on('message', (message) => {
      if (message.type === 'eth_subscription') {
        const { data } = message;
        if (data.subscription === subscriptionId) {
          if ('result' in data && typeof data.result === 'object') {
            const block = data.result;
            console.log(`New block ${block.number}:`, block);
          } else {
            console.error(`Something went wrong: ${data.result}`);
          }
        }
      }
    });
  })
  .catch((error) => {
    console.error(
      `Error making newHeads subscription: ${error.message}.
       Code: ${error.code}. Data: ${error.data}`
    );
  });
```

This would mean `messageListener` is never called with an `err`?